src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/CachedCompletedTask.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+
+namespace System.Threading.Tasks
+{
+    /// <summary>
+    /// Encapsulates the logic of caching the last synchronously completed task.
+    /// Used in classes like <see cref="MemoryStream"/> to reduce allocations.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
```

Comment:
can `T` be ever something different than an `int`? If not, we should rather make it non-generic and just use `Task<int>`. It would simplify the code and potentially prevent misusage (example: we most probably should not be caching `string` like that).

Reply:
Fixed. The type is now non-generic and supports only `Task<int>`.